merge_sort/merge_sort.js

```diff
@@ -0,0 +1,51 @@
+function merge (left, right) {
+	/*
+	Merging two sorted array
+	Time complexity : O(size)
+	Space complexity : O(size)
+	*/
+	let i = 0;
+	let j = 0;
+	let size = left.length + right.length;
+	let arr = [];
+	for (let k = 0; k < size; k++) {
+		if (i > left.length - 1 && j < right.length) {
+			arr[k] = right[j];
+			j++;
+		} else if (j > right.length - 1 && i < left.length) {
+			arr[k] = left[i];
+			i++;
+		} else if (left[i] < right[j]) {
+			arr[k] = left[i];
+			i++;
+		} else {
+			arr[k] = right[j];
+			j++;
+		}
+	}
+	return arr;
+}
+
+function mergeSort (arr) {
+	/*
+	Dividing the array into two parts, sorts both the part of array and then merge it
+	Time complexity : O(size(log(size)))
+	Space complexity : O(size)
+	*/
+	if (arr.length < 2) {
+		return arr;
+	}
+	let middle = arr.length / 2;
```

Comment:
middle value evaluates to a decimal value for arrays of odd length which does not seem right, use `//` i.e. strict division to return an integer off it.
Rest all looks good.